src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Automation/VsBuildManager.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.ProjectSystem.VS.ConnectionPoint;
+using Microsoft.VisualStudio.ProjectSystem.VS.Interop;
+using Microsoft.VisualStudio.Shell;
+using VSLangProj;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Automation
+{
+    /// <summary>
+    /// Undocumented.
+    /// </summary>
+    [Export(typeof(BuildManager))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    [Order(Order.Default)]
+    internal class VsBuildManager : ConnectionPointContainer,
+                                    IEventSource<_dispBuildManagerEvents>,
+                                    BuildManager,
+                                    BuildManagerEvents
```

Comment:
This needs to be returns by VSProject.Events.BuildManagerEvents somehow.

Reply:
We have `VisualBasicVSProjectEvents` which is easy to hook this into and fills the need. Is there a good reason not to just remove the VisualBasic capability check from that? It doesn't have anything interesting in it that I can see, I assume its there mainly to allow our `VisualBasicVSImports` to be imported, but that has the capability check so shouldn't be a problem.

Reply:
It was added to support "ImportsEvents"; but assuming VisualBasicVSImports continues to only apply to VB than that should be okay.

Reply:
I can't see why this pattern from VsProject.cs wouldn't work. I'll give it a go.

https://github.com/dotnet/project-system/blob/c2e466aabd03b83de4a3b0f44a488cc6d13ce790/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Automation/VsProject.cs#L69-L83